src/ir/predicate_textproto_to_rule_body_testdata.h

```diff
@@ -0,0 +1,122 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+#ifndef SRC_IR_PREDICATE_TEXTPROTO_TO_RULE_BODY_TESTDATA_H_H_
+#define SRC_IR_PREDICATE_TEXTPROTO_TO_RULE_BODY_TESTDATA_H_H_
+
+#include "absl/strings/substitute.h"
+
+namespace raksha::ir {
+// Note: StrFormat is careful not to allow dynamically constructed format
+// strings. In the few cases that it does allow format strings to come from
+// dynamic construction, it forces them to have a type indicating the format
+// specifiers at static time through a template. Instead, we use Substitute,
+// which is better suited for the purpose that we want. Therefore, access
+// path substitution points are indicated with $0 instead of %s.
+static constexpr std::tuple<absl::string_view, absl::string_view>
+predicate_textproto_to_rule_body_format[] = {
+    // Each of the tests is preceded by a snippet of what the predicate
+    // might look like in an Arcs manifest file, with the AccessPath
+    // represented by ap.
+
+    // ap is tag1
```

Comment:
Thanks for reformatting this. This looks much more readable!

Reply:
You're welcome! I agree, definitely an improvement.